### PR TITLE
Multiple new guild related features (and a bugfix)

### PR DESCRIPTION
**Pull request by kristofbolyai**
List of new features:
- Ability to recolor successful guild war related info to green (when the player's guild took a territory or successfully defended it)
- Added support for guild objective overlay (the commit (https://github.com/Wynntils/Wynntils/commit/89e3ccacaf1b06180b8ead7b161a824bc03d1a0e) also includes a fix to a bug where the index of the last scoreboard line displaying objective tasks was not calculated correctly, causing spam in the logger)
- Added an option to filter guild resource warnings from chat (disabled by default)
- Updated multiple guild related command aliases (read https://github.com/Wynntils/Wynntils/commit/e93bab14624cc86bcdf092336e52dee4dfbe3049 for more info) 
- Added an overlay to the guild manage menu so territories can be highlighted with a search box
- Added the ability to disable guild war times from the scoreboard (disabled by default)
- Added the ability to replace stars in guild chat names and use proper guild role names (`**** -> Chief`, `** -> Captain`)

### CODE REVIEW

**Review by Incompleteusern — APPROVED** [review]
LGTM!

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/GuildTerritoryManageOverlay.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            searchField.mouseClicked(offsetMouseX, offsetMouseY, e.getMouseButton());
+            if (e.getMouseButton() == 0) { // left click
+                if (searchField.isFocused()) {
+                    searchField.setCursorPositionEnd();
+                    searchField.setSelectionPos(0);
+                } else {
+                    searchField.setSelectionPos(searchField.getCursorPosition());
+                }
+            }
+        }
```
Comment (HeyZeer0) [inline src/main/java/com/wynntils/modules/utilities/overlays/inventories/GuildTerritoryManageOverlay.java]: This looks an "if hell" that can be easily cleaned up, whenever you are checking for a linear inverted condition use returns to make everything keep in the same line! Also perhaps most of this logic could be done inside the mouseClicked on the `searchField`! Be careful with this code suggestion tho, I made it on the web interface so spacing might be wrong 🤔 

```suggestion
        if (searchField == null) {
            return;
        }
        
        // Process the search field click interaction
        searchField.mouseClicked(offsetMouseX, offsetMouseY, e.getMouseButton());
        
        // Only listen for left clicks
        if (e.getMouseButton() != 0) {
            return;
        }
        
        if (searchField.isFocused()) {
            searchField.setCursorPositionEnd();
            searchField.setSelectionPos(0);
            return;
        }
        
        searchField.setSelectionPos(searchField.getCursorPosition());
```

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/GuildTerritoryManageOverlay.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            e.setCanceled(true);
+            if (e.getKeyCode() == Keyboard.KEY_ESCAPE) {
+                searchField.setFocused(false);
+            } else {
+                searchField.textboxKeyTyped(e.getTypedChar(), e.getKeyCode());
+            }
+        } else if (e.getKeyCode() == Keyboard.KEY_ESCAPE || e.getKeyCode() == McIf.mc().gameSettings.keyBindInventory.getKeyCode()) { // bank was closed by player
+            searchField = null;
+            inTerritoryManageMenu = false;
+        }
```
Comment (HeyZeer0) [inline src/main/java/com/wynntils/modules/utilities/overlays/inventories/GuildTerritoryManageOverlay.java]: These conditions can also be cleaned!

**Review by HeyZeer0 — CHANGES_REQUESTED** [review]
Awesome work! Looks good to me after the requested changes are done.
Also, do you mind embedding what the menu looks like with the overlay? 😄

**Comment by kristofbolyai** [comment]
> Also, do you mind embedding what the menu looks like with the overlay? 😄

image

Currently this overlay is really basic, but can be improved easily later on if suggestions come up.

**Pull request merged**